parcels/collection/collectionnodes.py

```diff
@@ -0,0 +1,1782 @@
+                        init_field = v.initial
+                        init_field.fieldset.computeTimeChunk(init_time, 0)
+                        setattr(pdata, v.name, init_field[init_time, pdata.depth, pdata.lat, pdata.lon])
+                        logger.warning_once("Particle initialisation from field can be very slow as it is computed in scipy mode.")
+                        # ==== ==== ==== #
+                        ndata.set_data(pdata)
+                        ndata = ndata.next
+                        i += 1
+                if v not in initialised:
+                    initialised.add(v)
+        else:
+            raise ValueError("Latitude and longitude required for generating ParticleSet")
+
+        self._ncount = len(self._data)
+        # ==== fill c-pointer ==== #
+        if self._ptype.uses_jit:
+            self._data_c = []
+            for i in range(len(self._data)):
+                node = self._data[i]
+                self._data_c.append(node.data.get_cptr())
+
+        self._iterator = None
+        self._riterator = None
+
+    def __del__(self):
+        """
+        Collection - Destructor
+        """
+        if self._data is not None and isinstance(self._data, DoubleLinkedNodeList):
+            del self._data
+        self._data = None
+        if self._data_c is not None:
+            del self._data_c
+        self._data_c = None
+        super(ParticleCollectionNodes, self).__del__()
+
+    def iterator(self):
+        """
+        :returns ParticleCollectionIterator, used for a 'for'-loop, in a forward-manner
+        """
+        self._iterator = ParticleCollectionIteratorNodes(self)
+        return self._iterator
+
+    def __iter__(self):
+        """
+        Returns an Iterator that allows for forward iteration over the
+        elements in the ParticleCollection (e.g. `for p in pset:`).
+        """
+        return self.iterator()
+
+    def reverse_iterator(self):
+        """
+        :returns ParticleCollectionIterator, used for a 'for'-loop, in a backward-manner
+        """
+        self._riterator = ParticleCollectionIteratorNodes(self, True)
+        return self._riterator
+
+    def __reversed__(self):
+        """Returns an Iterator that allows for backwards iteration over
+        the elements in the ParticleCollection (e.g.
+        `for p in reversed(pset):`).
+        """
+        return self.reverse_iterator()
+
+    def __getitem__(self, index):
+        """
+        Access a particle in this collection using the fastest access
+        method for this collection - by its index.
+
+        :arg index: int or np.int32 index of a particle in this collection
+        """
+        return self.get_single_by_index(index)
+
+    def __getattr__(self, name):
+        """
+        Access a single property of all particles.
+        CAUTION: this function is not(!) in-place and is REALLY slow
```

Comment:
Should this be a warning message whenever someone uses this function? (like how we throw a warning when a user initialises from a `Field`)